app/controllers/timelogs_controller.rb

```diff
@@ -0,0 +1,61 @@
+class TimelogsController < ApplicationController
+  before_action :load_user
+
+  def index
+    @timelog = Timelog.where(user_id: current_user.id)
+  end
+
+  def create
+    @timelog = @user.timelogs.create(timelog_params)
+    if @timelog.save
+      redirect_to user_timelogs_path(@user)
+    end
+
+  end
+
+  def update
+    @timelog = Timelog.find(params[:id])
+    if params[:timelog][:duration] == ""
+      params[:timelog][:duration] = 0
```

Comment:
I think we should add validation for timelog to be > 0. In other case, it doesn't make sense to store timelog with 0 duration.
In your case, it is a good practice to set `default` value on the sql level inside migration.